convert2rhel/subscription.py

```diff
@@ -286,6 +288,22 @@ def args(self):
 
         return args
 
+    @property
+    def connection_opts(self):
+        connection_opts = {}
+
+        if self.server_url:
+            url_parts = urllib.parse.urlsplit(self.server_url, allow_fragments=False)
+            connection_opts = {"host": url_parts.hostname}
+            connection_opts = {"port": url_parts.port}
+
+            path = urllib.parse.urlunsplit(
+                (None, None, url_parts.path, url_parts.query, url_parts.fragment))
+            connection_opts = {"handler": path}
```

Comment:
hm why not simply use `url_parts.path`? it is already `/path` as needed for "handler"

Reply:
So.... I was thinking that if query and fragment were specified in a url, the handler would be the place for those to be passed to the subscription-manager backend.
* Are they just not used in subscription-manager right now?  If so I think this is future-proofing so I think I'd keep it.
* Are they handled by some other connection_opt?  Let me know if so and I'll move them there.

Reply:
AFAICT, query and fragment are ignored right now. They are ignored after parsing URLs from the command lines, and there are no config knobs for them (sub-man does REST calls, so they would need to be passed to each).

Because of that, I'd simply ignore query and fragment altogether, as (if they would be ever needed) they would require additional work in sub-man.

Reply:
This has been updated to follow ptoscano's suggestion.  Rationale here: https://github.com/oamg/convert2rhel/pull/540#discussion_r943661175